src/Tokenizer/Tokens.php

```diff
@@ -565,25 +589,49 @@ public function getPrevNonWhitespace(int $index, ?string $whitespaces = null): ?
     }
 
     /**
-     * Get index for closest previous token of given kind.
+     * Get index for closest previous token among given kinds.
+     *
      * This method is shorthand for getTokenOfKindSibling method.
      *
-     * @param int $index token index
-     * @param list<array{int}|string|Token> $tokens        possible tokens
-     * @param bool $caseSensitive perform a case sensitive comparison
+     * @param int                                          $index         token index
+     * @param list<array{0: int, 1?: string}|string|Token> $tokens        possible tokens
+     * @param bool                                         $caseSensitive perform a case sensitive comparison
      */
     public function getPrevTokenOfKind(int $index, array $tokens = [], bool $caseSensitive = true): ?int
     {
         return $this->getTokenOfKindSibling($index, -1, $tokens, $caseSensitive);
     }
 
+    /**
+     * Get index for closest previous token of given kind.
+     *
+     * @param int                                        $index         token index
+     * @param array{0: int, 1?: string}|int|string|Token $tokens        possible tokens
```

Comment:
case 2

from docs, we allow multiple token kinds, from code - only one.
can we run `getPrevTokenOfKind` / `getNextTokenOfKind` analysis to see how many cases we have when we pass single token vs when we pass multiple tokens to match against ?

Reply:
> from docs, we allow multiple token kinds, from code - only one?

Actually the doc also indicates only one token is expected. The parameter name is wrong though.

> why we have this conditional conversion $tokens = [$tokens]; ?

Because if we look for a token id, the current method expectes it to be wrapped in an array.

> can we run `getPrevTokenOfKind` / `getNextTokenOfKind` analysis to see how many cases we have when we pass single token vs when we pass multiple tokens to match against ?

I could find more than 200 calls to both methods. Close to 60 of them provide multiple tokens to match against.

Reply:
> > from docs, we allow multiple token kinds
> 
> Actually the doc also indicates only one token is expected.

in docs, you put `$tokens        possible tokens`, which is plural [in both, description and variable name]

> I could find more than 200 calls to both methods. Close to 60 of them provide multiple tokens to match against.

If you would agree with goal to have only one method [that is throwing exception] and not one [that is throwing exception] and other [that is returning null], I believe we should keep possibility to pass multiple allowed token kinds